From stable-bounces@linux.kernel.org Thu Jan 19 09:30:47 2006
Date: Thu, 19 Jan 2006 17:29:52 +0000
From: Ralf Baechle DL5RB <ralf@linux-mips.org>
To: stable@kernel.org
Message-ID: <20060119172952.GA15302@linux-mips.org>
Content-Disposition: inline
Cc: 
Subject: Fix mkiss locking bug

ax_encaps() forgot to drop the bufferlock at the end of the function.
Patch is already in 2.6.16-rc1.

Signed-off-by: Ralf Baechle DL5RB <ralf@linux-mips.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/hamradio/mkiss.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.15.1.orig/drivers/net/hamradio/mkiss.c
+++ linux-2.6.15.1/drivers/net/hamradio/mkiss.c
@@ -515,6 +515,7 @@ static void ax_encaps(struct net_device 
 			count = kiss_esc(p, (unsigned char *)ax->xbuff, len);
 		}
   	}
+	spin_unlock_bh(&ax->buflock);
 
 	set_bit(TTY_DO_WRITE_WAKEUP, &ax->tty->flags);
 	actual = ax->tty->driver->write(ax->tty, ax->xbuff, count);
